pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -405,6 +405,10 @@ func (c *Controller) reconcileShoot(ctx context.Context, logger *logrus.Entry, g
 		reconcileAllowed                           = !failedOrIgnored && ((!reconcileInMaintenanceOnly && !confineSpecUpdateRollout(shoot.Spec.Maintenance)) || !isUpToDate || (isNowInEffectiveShootMaintenanceTimeWindow && !alreadyReconciledDuringThisTimeWindow))
 	)
 
+	if operationType == gardencorev1beta1.LastOperationTypeMigrate {
```

Comment:
As discussed in person: let's not rely on this check here and rather solve the general problem by the following approach:
in `updateShootStatusOperationStart` let's not retry the status update on conflict and instead directly return any error to requeue the shoot (with exponential backoff as usual) to make sure, the controller always starts with recent data and doesn't start any operation based on stale data.

/author-action